src/pages/groupComparison/MRNAEnrichments.tsx

```diff
@@ -23,14 +24,28 @@ export default class MRNAEnrichments extends React.Component<IMRNAEnrichmentsPro
     }
 
     private readonly enrichmentAnalysisGroups = remoteData({
-        await: () => [this.props.store.activeGroups],
+        await: () => [this.props.store.activeGroups, this.props.store.sampleSet],
         invoke: () => {
-            const groups = this.props.store.activeGroups.result!.map(group => ({
-                name: group.nameWithOrdinal,
-                description: `samples in ${group.nameWithOrdinal}`,
-                count: getNumSamples(group),
-                color: group.color
-            }));
+            const sampleSet = this.props.store.sampleSet.result || new ComplexKeyMap<Sample>();
```

Comment:
now that there's more logic it is more glaring that there's so much code duplication in MRNAEnrichments::enrichmentAnalysisGroups and ProteinEnrichments::enrichmentAnalysisGroups